### ISSUE TITLE
Delay autokarma pushes on old branches until new branches have appropriate requests

### ISSUE THREAD

**Issue by bowlofeggs** [fedora-infra/bodhi#1661]
This issue was [originally filed](https://bugzilla.redhat.com/show_bug.cgi?id=1281536) in Red Hat's Bugzilla by Andy Lutomirski:

> 
> Suppose that foo-1.0-1 is available in F22, F23, and rawhide.  If I build foo-1.0-2.fc22, foo-1.0-2.fc23, and foo-1.0-2.fc24, rawhide will by updated in the next compose.  If I submit updates with autokarma enabled for f22 and f23 and only foo-1.0-2.fc22 gets sufficient karma, then f22 stable will be requested and, once it's composed and mirrored out, the upgrade path to f23 will be broken until either foo-1.0-2.fc23 gets enough karma or the packager manually requests stable.  This could take an unbounded amount of time.
> 
> Please consider adding a Bodhi feature that will cause all autokarma pushes on a branch to wait until a new build (i.e. same or newer NVR) exists for rawhide and all newer branches *and* has been requested for stable in all newer branches.
> 
> In an ideal world, composes would be instantaneous or synchronized and we could make sure that there's never a state in which the old branch is composed first, but, in the mean time, simply making autokarma insist on tagging the builds in the correct order would avoid frequent long periods of time in which the upgrade path is broken.

Sergio Monteiro Basto further commented that Taskotron could help with this (it does test for it, and @kparal has recently been working on getting Bodhi's backend to gate based on Taskotron results):

> Taskotron could help us, for example in [1] we can read : "error: Condition not satisfied: pngquant-2.5.2-5.fc22 <= pngquant-2.5.2-4.fc24", which reminder me that I forgot the build it for rawhide . 
> 
> [1]
> https://taskotron.fedoraproject.org/artifacts/all/c06ee9d4-acf6-11e5-bcf5-52540053ee00/task_output/FEDORA-2015-4a647a067b.log

**Comment by kparal** [comment]
Yes, that would be a great feature. Maintainers should probably be able to override that though, e.g. during periods of Branched freezes (otherwise updates would be frozen even for stable releases), or for security updates.
